examples/pytorch/gat/main.py

```diff
@@ -93,22 +74,26 @@ def train_func(config: Dict):
     )
 
     address = "localhost:9999"
-    s = Server(address, args.data_dir, 0, len(args.partitions))
+    s = Server(address, config["data_dir"], 0, config["partitions"])
     g = DistributedClient([address])
-    max_id = g.node_count(args.node_type) if args.max_id in [-1, None] else args.max_id
-    dataset = ray.data.range(max_id).repartition(max_id // args.batch_size)
-    pipe = dataset.window(blocks_per_window=4).repeat(args.num_epochs)
+    dataset = ray.data.read_text(config["sample_file"])
+    dataset = dataset.repartition(dataset.count() // config["batch_size"])
+    pipe = dataset.window(blocks_per_window=4).repeat(config["num_epochs"])
 
     def transform_batch(idx: list) -> dict:
         return model.q.query_training(g, np.array(idx))
 
     pipe = pipe.map_batches(transform_batch)
 
-    for epoch in range(epochs_trained, args.num_epochs):
+    for epoch, epoch_pipe in enumerate(pipe.iter_epochs()):
+        if epoch < epochs_trained:
```

Comment:
Not needed. Please, follow the pyg example in ray: training loop and then evaluation a test/validation dataset.

Reply:
The "if epoch < epochs_trained:" is not needed? If so removed. The other line is the equivalent of for epoch in range(n_epochs) of that example.